src/recycle/shared_pool.hpp

```diff
@@ -288,6 +320,9 @@ class shared_pool
         /// threads releases a resource into the free list while
         /// another tries to read its size.
         mutable mutex_type m_mutex;
+
+        /// The number of items allocated so far
+        std::atomic<uint32_t> m_num_allocated;
```

Comment:
Lets call it something like `uint32_t m_allocated_resources` since we call it resources everywhere.